File: problems/projection_area_of_3d_shapes/solution.py
class Solution:
    def projectionArea(self, grid: List[List[int]]) -> int:
        top, front, side = len(grid) ** 2, 0, 0
        for arr in grid:
            for cell in arr:
                if cell <= 0:
                    top -= 1
        for arr in grid:
            front += max(arr)
        sides = list(zip(*grid))
        for tup in sides:
            side += max(tup)
        return top + front + side

File: problems/projection_area_of_3d_shapes/solution.js
const projectionArea = function(grid) {
    let top = 0
    let front = 0, side = 0
    let sides = Array(grid.length).fill(0)
    for(const set of grid){
        front += Math.max(...set)
    }
    for(const set of grid){
        for(t = 0; t < set.length; t++){
            if(set[t]) top += 1
            sides[t] = Math.max(sides[t], set[t])
        }
    }
    sides = sides.reduce((a,b) => a + b)
    return top + front + sides
}